src/api/app/controllers/webui/request_controller.rb

```diff
@@ -79,7 +115,46 @@ def modify_review
     redirect_to request_show_path(number: request), success: 'Successfully submitted review'
   end
 
+  def sort_show_events
+    events = @bs_request.comments.without_parent.includes([:user, { children: :user }]).to_a
+    events |= @bs_request.request_history_elements.to_a
+    events |= @bs_request.reviews.to_a
+    events.sort_by!(&:created_at)
+    # sort out doubled events
+    @events = []
+    reviews_seen = []
+    events.each do |event|
+      if event.is_a?(Review)
+        next if reviews_seen.include?(event.id)
+        reviews_seen << event.id
+      end
+      if event.is_a?(HistoryElement::RequestReviewAdded)
+        next if reviews_seen.include?(event.description_extension.to_i)
+        reviews_seen << event.description_extension.to_i
+      end
+      @events << event
+    end
+  end
+
```

Comment:
why a method in the controller? Could you put in a service object  (i.e EventSorterService) and write tests for it?